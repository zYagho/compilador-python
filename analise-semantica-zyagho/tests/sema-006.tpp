{Aviso: Variável 'b' declarada e inicializada, mas não utilizada}
{Aviso: Variável 'c' declarada e não inicializada}
{Aviso: Variável 'a' declarada e inicializada, mas não utilizada}
{Aviso: Atribuição de tipos distintos. Coerção implícita do valor de 'b' é do tipo 'inteiro' para 'c' que é 'flutuante'}
{Aviso: Atribuição de tipos distintos. Coerção implícita do valor retornado por 'func' é do tipo 'inteiro' para 'a' que é 'flutuante'}
{Erro: Função 'principal' deveria retornar 'inteiro', mas retorna 'vazio'}

inteiro func(inteiro: x, inteiro: y)
	retorna (x + y)
fim

inteiro principal()
	flutuante: a
	flutuante: c
	inteiro: b

	b := c

	a := func(10,5)
fim
