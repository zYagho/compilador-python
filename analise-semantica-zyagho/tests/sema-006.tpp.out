WAR-SEM-VAR-DECL-NOT-USED
WAR-SEM-IMP-COERC-OF-VAR
WAR-SEM-IMP-COERC-OF-RET-VAL
ERR-SEM-FUNC-RET-TYPE-ERROR
